packages/ui/components/core/src/SlotMixin.js

```diff
@@ -6,10 +6,145 @@ import { isTemplateResult } from 'lit/directive-helpers.js';
 /**
  * @typedef {import('../types/SlotMixinTypes.js').SlotMixin} SlotMixin
  * @typedef {import('../types/SlotMixinTypes.js').SlotsMap} SlotsMap
+ * @typedef {import('../types/SlotMixinTypes.js').SlotFunctionResult} SlotFunctionResult
+ * @typedef {import('../types/SlotMixinTypes.js').SlotRerenderObject} SlotRerenderObject
  * @typedef {import('lit').LitElement} LitElement
  */
 
+const isRerenderConfig = (/** @type {SlotFunctionResult} */ o) =>
+  !Array.isArray(o) && typeof o === 'object' && 'template' in o;
+
 /**
+ * # Systems >> Core >> SlotMixin ||20
```

Comment:
Doesn't make sense to have the documentation double, keep 1 source of truth.

Reply:
Yeah agree with SSOT. Also like the principle of having docs as close to code as possible (will be better maintained and read by devs using code base), but we don't have a build process that allows for this (like angularjs had back in the days).
Compromise: I add a short summary and mention the existence of the extended docs